scheduler/context.go

```diff
@@ -72,9 +77,17 @@ func (e *EvalCache) SemverConstraintCache() map[string]VerConstraints {
 	return e.semverCache
 }
 
+type EvalEvent struct {
+	Reason      string
+	NetIndex    *structs.NetworkIndex
```

Comment:
I'm hesitant to say we should include this because it's going to be big (8KB per IP for Bitmaps), and we should be able to regenerate it from the Node and Allocations fields.

However, since we believe this code is rarely hit and there may be bugs in the network index itself, let's do it.

Allocation copying is probably going to be more expensive, but we definitely need to do it.

Reply:
Yeah, I was on the fence about it, but since this is such a hard bug to track I figured it would good to have this information, even as way to rule out `NetIndex`. 

I can add a comment to reevaluate the need for this field in the future.